### PR TITLE
Improvements: Swapped source and destination in time series graph for non-directed edges

### PR DESCRIPTION
**Pull request by matteo-causalens**
## Description and Motivation

Modified `cai_causal_graph.time_series_causal_graph.TimeSeriesCausalGraph.add_edge` in 
`cai_causal_graph.time_series_causal_graph.TimeSeriesCausalGraph` to always order source and destination nodes by
`time_lag` even when the edge is not directed. For example, `add_edge('c', 'a (lag=1)', edge_type='--')` will add an 
undirected edge `a (lag=1) -- c` instead of `c -- a (lag=1)`. This is done just for convenience and to avoid confusion.

## How Has This Been Tested?
unit tests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Improvement (non-breaking change which improves functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactor (moving code around, general refactoring improvements)
- [ ] Documentation (documentation)

## PR Checklist:

- [x] I have reviewed my own PR? (review the PR as you are reviewing someone else's PR)
- [] I have implemented all requirements? (see JIRA, project documentation)
- [ ] I am affecting someone else's work? If so: I have added those people as reviewers?
- [ ] I have added comments to all the bits that are hard to follow
- [ ] At a bare minimum, I tested this manually
- [ ] I have added unit test(s)
- [ ] Is this a bugfix? If so have I added a regression test?
- [ ] Does this PR satisfy the [one PR, one concern](https://medium.com/@fagnerbrack/one-pull-request-one-concern-e84a27dfe9f1) rule?
- [ ] If needed, documentation has been added/updated?
- [ ] I have updated the appropriate changelog with a line for my changes
- [ ] If this PR breaks reproducibility, have I added a note in the changelog explaining the break in reproducibility
      and its extent?

### CODE REVIEW

**Review by ilya-causalens — APPROVED** [review]

File: changelog.md [hunk 1]
```diff
@@ -1,5 +1,12 @@
 # Changelog
 
+## NEXT
+
+- Modified `cai_causal_graph.time_series_causal_graph.TimeSeriesCausalGraph.add_edge` in
+  `cai_causal_graph.time_series_causal_graph.TimeSeriesCausalGraph` to always order source and destination nodes by
+  `time_lag` even when the edge is not directed. For example, `add_edge('c', 'a (lag=1)', edge_type='--')` will add a
```
Comment (andrew-causalens) [inline changelog.md:7]: Use enumeration for edge type. Don't want to recommend that the user uses string.

Comment (andrew-causalens) [inline changelog.md:7]: "a undirected" -> "an undirected"

File: cai_causal_graph/time_series_causal_graph.py [hunk 2]
```diff
@@ -811,19 +811,49 @@ def add_edge(
...
+                elif isinstance(source, str):
+                    source_time_lag = get_variable_name_and_lag(source)[1]
+                else:
+                    assert isinstance(source, TimeSeriesNode)   # for linting
+                    source_time_lag = source.time_lag
+
+                if isinstance(destination, Node) and not isinstance(destination, self._NodeCls):
+                    tmp_destination = self._NodeCls.from_dict(destination.to_dict())
+                    destination_time_lag = tmp_destination.time_lag
+
```
Comment (andrew-causalens) [inline cai_causal_graph/time_series_causal_graph.py:846]: No need for this white space.

File: cai_causal_graph/time_series_causal_graph.py [hunk 3]
```diff
@@ -811,19 +811,49 @@ def add_edge(
...
+                if edge_type != EdgeType.DIRECTED_EDGE:
+                    # if time_lag for source is greater than destination, then swap source and destination
+                    assert isinstance(source, TimeSeriesNode) and isinstance(destination, TimeSeriesNode)
+                    if source_node.time_lag > destination_node.time_lag:
+                        destination_node, source_node = source_node, destination_node
                 edge = self._EdgeCls(
                     source_node, destination_node, edge_type=edge.get_edge_type(), meta=edge.get_metadata()
                 )
 
+        else:
```
Comment (andrew-causalens) [inline cai_causal_graph/time_series_causal_graph.py:832]: Can you just add a comment here so user knows what is happening. Lots of new code and essentially just putting the nodes in the right format. But just explain this else block a bit.

**Review by andrew-causalens — APPROVED** [review]
Looks good. Just a little minor cleanup.

**Pull request merged**